eunkyung/app.js

```diff
@@ -0,0 +1,53 @@
+const http = require('http')
+const express = require('express')
+const cors = require('cors')
+const dotenv = require('dotenv')
+const morgan = require('morgan')
+
+dotenv.config()
+
+const { DataSource } = require('typeorm')
+
+const myDataSource = new DataSource({
+    type: process.env.TYPEORM_CONNECTION,
+    host: process.env.TYPEORM_HOST,
+    port: process.env.TYPEORM_PORT,
+    username: process.env.TYPEORM_USERNAME,
+    password: process.env.TYPEORM_PASSWORD,
+    database: process.env.TYPEORM_DATABASE
+})
+
+myDataSource.initialize()
+    .then(() => { console.log('data server has been initialized!') })
+
+const app = express()
+
+app.use(express.json())
+app.use(cors())
+app.use(morgan('tiny'))
+
+app.get('/ping', (req, res) => {
+    res.status(200).json({ message: 'pong' })
+})
+
+app.post('/sign_up', async (req, res, next) => {
+    const { name, password, email } = req.body
+    await myDataSource.query(
+        `INSERT INTO users (
+            name, 
+            password, 
+            email) 
+            VALUES (?,?,?);`,
+        [name, password, email]
+    )
+    res.status(201).json({ message: 'userCreated' })
+})
+
+
+
+const server = http.createServer(app)
+const PORT = process.env.PORT
+
+const start = async () => { server.listen(PORT, () => { console.log(`server is listening on ${PORT}`) }) }
```

Comment:
- 이 부분도 가독성 및 유지 보수 측면에서 코드를 아래와 같이 수정해주면 더 좋을 것 같습니다:
- http 모듈의 createServer함수를 통해 서버를 구축하는 것이 아닌, express() 함수가 할당된 app 변수를 사용하여 서버를 실행해주는 코드로 수정해주세요.

```suggestion
const start = async () => {
  app.listen(PORT, () => {
    console.log(`server is listening on ${PORT}`)
  })
}
```